rusk-wallet/src/lib/config.rs

```diff
@@ -0,0 +1,243 @@
+
+    use crate::Error;
+    use serde::Deserialize;
+    use std::path::PathBuf;
+
+    #[derive(Deserialize)]
+    pub struct ParsedConfig {
+        pub wallet: ParsedWalletConfig,
+        pub rusk: ParsedRuskConfig,
+        pub explorer: ParsedExplorerConfig,
+    }
+
+    #[derive(Deserialize)]
+    pub struct ParsedWalletConfig {
+        pub data_dir: Option<PathBuf>,
+        pub wallet_name: Option<String>,
+        pub wallet_file: Option<PathBuf>,
+        pub skip_recovery: Option<bool>,
+    }
+
+    #[derive(Deserialize)]
+    pub struct ParsedRuskConfig {
+        pub ipc_method: Option<String>,
+        pub rusk_addr: Option<String>,
+        pub rusk_port: Option<String>,
+        pub prover_addr: Option<String>,
+        pub prover_port: Option<String>,
+        pub socket_path: Option<String>,
+    }
+
+    #[derive(Deserialize)]
+    pub struct ParsedExplorerConfig {
+        pub tx_url: Option<String>,
+    }
+
+    /// Attempts to parse the content of a file into config values
+    pub fn parse(content: &str) -> Result<ParsedConfig, Error> {
+        toml::from_str(content).map_err(Error::TOML)
+    }
+}
+
+/// Config holds the settings for the CLI wallet
+#[derive(Debug)]
+pub(crate) struct Config {
+    /// Wallet configuration
+    pub wallet: WalletConfig,
+    /// Rusk connection configuration
+    pub rusk: RuskConfig,
+    /// Dusk explorer configuration
+    pub explorer: ExplorerConfig,
+}
+
+/// Wallet and store configuration
+#[derive(Debug)]
+pub(crate) struct WalletConfig {
+    /// Directory to store user data
+    pub data_dir: PathBuf,
+    /// Wallet file name
+    pub name: String,
+    /// Path to a wallet file. Overrides `data-dir` and `wallet-name`
+    pub file: Option<PathBuf>,
+    /// Skip wallet recovery phrase (useful for headless wallet creation)
+    pub skip_recovery: bool,
+}
+
+/// Connection details to Rusk and Prover clusters
+#[derive(Debug)]
+pub(crate) struct RuskConfig {
+    /// IPC method for communication with rusk
+    pub ipc_method: String,
+    /// Rusk address
+    pub rusk_addr: String,
+    /// Rusk port
+    pub rusk_port: String,
+    /// Prover service address
+    pub prover_addr: String,
+    /// Prover service port
+    pub prover_port: String,
+    /// Path for setting up the unix domain socket
+    pub socket_path: String,
+}
+
+/// Dusk Explorer access information
+#[derive(Debug)]
+pub(crate) struct ExplorerConfig {
+    /// Base url for transactions
+    pub tx_url: Option<String>,
+}
+
+impl Config {
+    /// Attempt to load configuration from file
+    pub fn load() -> Option<Config> {
+        // search in default data dir and current working directory
+        let mut paths =
+            vec![env::current_dir().ok()?, LocalStore::default_data_dir()];
+
+        for p in paths.iter_mut() {
+            // file is always called "config.toml"
+            p.push("config");
+            p.set_extension("toml");
+            println!("{}", p.display());
+
```

Comment:
Its unclear why we `println` here. Maybe use [tracing](https://crates.io/crates/tracing) (or some other configurable log environment) instead?

Reply:
Woops, that one slipped, good catch!